svgpathtools/path.py

```diff
@@ -3325,3 +3346,161 @@ def _parse_path(self, pathdef, current_pos=0j, tree_element=None):
+                other.origin.y + other.height < self.origin.y:
+            return False
+        return True
+
+    def to_tuple(self):
+        return self.origin.x, self.origin.y, self.width, self.height
+
+
+class QuadTree:
+    """
+    QuadTree implementation adapted to SVG paths for faster intersection lookups
+    """
+    def __init__(self, boundary: Rect, capacity=10):
+        self.boundary = boundary
+        self.capacity = capacity
+        self._path_segments = []
+        self._is_split = False
+        self._subtreeNE = None
+        self._subtreeNW = None
+        self._subtreeSE = None
+        self._subtreeSW = None
+
+    def _split(self):
+        self._subtreeNE = QuadTree(
+            Rect(self.boundary.origin,
+                 int(self.boundary.width / 2), int(self.boundary.height / 2)),
+            self.capacity
+        )
+        self._subtreeNW = QuadTree(
+            Rect(self.boundary.origin.translated(complex(self.boundary.width / 2, 0)),
+                 int(self.boundary.width / 2), int(self.boundary.height / 2)),
+            self.capacity
+        )
+        self._subtreeSE = QuadTree(
+            Rect(self.boundary.origin.translated(complex(0, self.boundary.height / 2)),
+                 int(self.boundary.width / 2), int(self.boundary.height / 2)),
+            self.capacity
+        )
+        self._subtreeSW = QuadTree(
+            Rect(self.boundary.origin.translated(complex(self.boundary.width / 2, self.boundary.height / 2)),
+                 int(self.boundary.width / 2), int(self.boundary.height / 2)),
+            self.capacity
+        )
+        self._is_split = True
+
+    def insert_segment(self, segment: PathSegment):
+        """will insert a PathSegment into the tree"""
+        if not self.boundary.overlaps(bbox_to_rect(*segment.original_segment.bbox())):
+            return
+
+        if len(self._path_segments) < self.capacity:
+            self._path_segments.append(segment)
+        else:
+            if not self._is_split:
+                self._split()
+
+            self._subtreeNE.insert_segment(segment)
+            self._subtreeNW.insert_segment(segment)
+            self._subtreeSE.insert_segment(segment)
+            self._subtreeSW.insert_segment(segment)
+
+    def insert_path(self, path: Path):
+        """will split the path into segments and call insert_segment on each one"""
+        for segment in path:
+            self.insert_segment(PathSegment(segment, original_path=path))
+
+    def _get_segments_in_area(self, area: Rect, out=None) -> list[PathSegment]:
+        """returns a list of PathSegments that are present in the specified area"""
+        if out is None:
+            out = list()
+
+        if not self.boundary.overlaps(area):
+            return out
+
+        out.extend(self._path_segments)
+        if self._is_split:
+            out = out.union(self._subtreeNE._get_segments_in_area(area, out))
```

Comment:
out is a `list()` there is no method `.union`. for list, or at least not in 3.8 which I run.

https://www.programiz.com/python-programming/methods/list